server/datastore/mysql/policies.go

```diff
@@ -1832,14 +1832,42 @@ func (ds *Datastore) getPoliciesBySoftwareTitleIDs(
 		tmID = *teamID
 	}
 
-	query, args, err := sqlx.In(query, softwareTitleIDs, softwareTitleIDs, tmID)
-	if err != nil {
-		return nil, ctxerr.Wrap(ctx, err, "build select get policies by software id query")
-	}
+	// // https://dev.mysql.com/doc/refman/8.4/en/packet-too-large.html
+	// maxAllowedPacket := 16 * 1024 * 1024 // 16MB / 16,777,216 bytes
+	// maxPlaceholders := 65535             // mysql max placeholders
+	// var x uint
+	// sizePerPlaceholder := int(unsafe.Sizeof(x)) // 64 bits / 8 bytes for unsigned integers
+	// placeholdersPerID := 2
+
+	// // Calculate the size of the base query
+	// baseQuerySize := len(baseQuery) // 397
+
+	// // maximum number of IDs that can fit into a single query given maxAllowedPacket size
+	// maxBatchSizeQuerySize := (maxAllowedPacket - baseQuerySize) / ((sizePerPlaceholder * placeholdersPerID) + 8) // 8 for the team ID = 699034
+	// // maximum number of IDs that can fit into a single query given maxPlaceholders
+	// maxBatchSizePlaceholders := (maxPlaceholders / placeholdersPerID) - 1 // -1 for the team ID = 32766
+	// batchSize := min(maxBatchSizePlaceholders, maxBatchSizeQuerySize)     // 32766
```

Comment:
I can leave this math in the code to show how I arrived at this number or if it's too messy can add it as comments in the PR. Dealers choice.

Reply:
Probably worthwhile to pull this into a comment on the issue (or updating To Fix on the bug) rather than putting it here, mainly because this comment will drift as we have to tweak limits.

Also, let's give ourselves more headroom on batching in case we add other filters and drop the batch size to an even 32,000 (or 32,500 if that feels like leaving too many placeholders on the table :) ). Perf impact won't be much and we won't have to fiddle with max batch size if we add another parameter in here.

Side note: thanks for doing what I didn't and using the batch-processing MySQL function on this, as we _should_ be using that in more places.